Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift

```diff
@@ -247,7 +247,7 @@ public class DocumentationContentRenderer {
         return true
     }
 
-    static func renderKindAndRole(_ kind: DocumentationNode.Kind?, semantic: Semantic?) -> (RenderNode.Kind, String) {
+    static func renderKindAndRole(_ kind: DocumentationNode.Kind?, semantic: Semantic? = nil, linkSummary: LinkDestinationSummary? = nil) -> (RenderNode.Kind, String) {
```

Comment:
If I understand this correctly, this method is not expected to receive both a semantic and a link summary because one represents a local page and the other represents an external page. If we need a version that accepts a link summary we could consider having either a separate overload (that calls into a private common implementation) or a parameter that can be either a Semantic or a LinkDestinationSummary.

Reply:
If it's not expected to pass both semantic and linkSummary at the same invocation call, let's split the method into two making them an overload.

The original 
```swift
 static func renderKindAndRole(_ kind: DocumentationNode.Kind?, semantic: Semantic?) -> (RenderNode.Kind, String) {
```
and the overload
```swift
 static func renderKindAndRole(_ kind: DocumentationNode.Kind?, linkSummary: LinkDestinationSummary?) -> (RenderNode.Kind, String) {
```

Then the common logic can be encapsulated in another function.